src/Files.App/Utils/Git/GitHelpers.cs

```diff
@@ -664,5 +706,33 @@ private static bool IsAuthorizationException(Exception ex)
 				ex.Message.Contains("status code: 401", StringComparison.OrdinalIgnoreCase) ||
 				ex.Message.Contains("authentication replays", StringComparison.OrdinalIgnoreCase);
 		}
+
+		private static void DisposeIfFinished()
+		{
+			if (Interlocked.Decrement(ref _activeOperationsCount) == 0)
+				TryDispose();
+		}
+
+		private static async Task PostMethodToThreadWithMessageQueueAsync(Action payload)
+		{
+			Interlocked.Increment(ref _activeOperationsCount);
+			_owningThread ??= new ThreadWithMessageQueue();
+
+			await _owningThread.PostMethod(payload);
+
+			DisposeIfFinished();
```

Comment:
```suggestion
			try
			{
				_owningThread ??= new ThreadWithMessageQueue();

				await _owningThread.PostMethod(payload);
			}
			finally
			{
				DisposeIfFinished();
			}
```
`DisposeIfFinished()` should always be executed.

Reply:
Different methods need different exception handling, I think we should call `DisposeIfFinished()` separately. The other option is to use some kind of error codes

Reply:
I think at least the decrementing of the count should be done in `finally` to maintain the integrity of the count.